src/Plugin/Block/SkipNavAnchorBlock.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Drupal\jumpstart_ui\Plugin\Block;
+
+use Drupal\Component\Utility\Html;
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Url;
+
+/**
+ * Provides a block with a hidden anchor.
+ *
+ * @Block(
+ *   id = "jumpstart_ui_skipnav_main_anchor",
+ *   admin_label = @Translation("Main content anchor target"),
+ * )
+ */
+class SkipNavAnchorBlock extends BlockBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $id = Html::getUniqueId('main-content');
+    return [
+      'anchor' => [
+        '#title' => $this->t('Main Content'),
+        '#type' => 'link',
+        '#url' => Url::fromUri('internal:#' . $id),
+        '#attributes' => [
+          'name' => $id,
+          'id' => $id,
+          'tabindex' => '-1',
+          'class' => ['visually-hidden', 'focusable'],
+        ],
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function blockAccess(AccountInterface $account) {
+    return AccessResult::allowedIfHasPermission($account, 'access content');
```

Comment:
What happens with `404` pages?

Reply:
I believe anonymous users can access 404 pages. What are you questioning here?

Reply:
Do you mean 403 pages?

Well, if that is the case and this block is placed there I suspect the fallback that is built into the templates would be what is used.  This block is really for changing that default anchor to somewhere within content where the layout is more complex.

Reply:
I'm just thinking if this is a skip nav anchor, wouldn't we want all users regardless of permissions to skip to the main content. the main content might be "Hey you're not allowed here" but they should be able to skip nav still.

Reply:
Yeah, makes sense. I figured I should have some sort of access check and `access content` seemed to be paired well. I guess we could just remove it and leave it wide open.